### PR TITLE
Add ability to run Docker from Docker container running Jenkins

### PR DESCRIPTION
**Pull request by MichaelRoytman**

### CODE REVIEW

**Comment by MichaelRoytman** [comment]
@e0d @jibsheet Here is the code to build the xqueue image.

File: docker-compose.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ edxops-jenkins:
   container_name: edxops-jenkins
   volumes:
       - .docker-volumes/jenkins:/var/jenkins_home
+      - /var/run/docker.sock:/var/run/docker.sock
```
Comment (e0d) [inline docker-compose.yml]: What is using this?

File: docker/Dockerfile [hunk 2]
```diff
@@ -3,8 +3,13 @@ MAINTAINER edxops
...
 USER root
 RUN apt-get update \
-      && apt-get install -y sudo python-dev python-pip libmysqlclient-dev \
+      && apt-get install -y sudo python-dev python-pip libmysqlclient-dev apt-transport-https ca-certificates \
       && rm -rf /var/lib/apt/lists/*
+RUN apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
+COPY docker.list /etc/apt/sources.list.d/docker.list
+RUN apt-cache policy docker-engine
+RUN sudo apt-get update
+RUN sudo apt-get install -y docker-engine
```
Comment (e0d) [inline docker/Dockerfile]: I wonder if we want to have a separate Dockerfile for this usecase.  I think, typically, we wouldn't need docker available in the container when developing DSL.

File: docker/docker-in-docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM jenkins:latest
+MAINTAINER edxops
+
+USER root
+RUN apt-get update \
+      && apt-get install -y sudo python-dev python-pip libmysqlclient-dev apt-transport-https ca-certificates \
+      && rm -rf /var/lib/apt/lists/*
+RUN apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
+COPY docker.list /etc/apt/sources.list.d/docker.list
+RUN apt-cache policy docker-engine
```
Comment (feanil) [inline docker/docker-in-docker/Dockerfile]: You don't need to run this as a part of the build.  It was just in the instructions for visual verification.

**Comment by feanil** [comment]
:+1:

**Pull request merged**